### DISCUSSION TITLE
Deploy to Deno

### DISCUSSION THREAD

**Discussion by johnd0e** [PublicAffairs/openai-gemini#19] [Announcements]
This process does not require any special support from the project.

There are several alternative methods:

- Via [Playground](https://docs.deno.com/deploy/manual/playgrounds/)
  This is the simplest method, but keep in mind that if there is a future update, you will need to update the code manually.
  1. Register/Login at https://dash.deno.com/.
  2. Press the big blue **New Playground** button.
  3. Enter the code from https://github.com/PublicAffairs/openai-gemini/blob/main/src/worker.mjs.
  4. Optionally, edit the title, as it will serve as your site address.
  5. Click the big blue **Save & Deploy** button.
- With Github integration.
  1. [Fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo#forking-a-repository) this repository.
  2. Register/Login at https://dash.deno.com/.
  3. Create a new project at https://dash.deno.com/new_project and link to your fork.
  4. Fill other required fields:
     a. Choose a name carefully, as it will be your site address
     b. Set the entry point to `src/worker.mjs`
  6. Click the big blue **Deploy Project** button.
    <details><summary>This is how it should look</summary>

   ![Screenshot_20-10-2024_22299_dash deno com](https://github.com/user-attachments/assets/277f23b8-6e39-40b5-9b89-df1b3158259d)

  </details>

  In order to update your site you need just to [sync](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork?platform=windows#syncing-a-fork-branch-from-the-web-ui) your fork.

- Using `deployctl` on the command line
  Refer to the  [docs](https://docs.deno.com/deploy/manual/deployctl/) for details.

**Comment by hinmer** [comment]
方法一报错：HttpError: 404 Not Found
    at Object.fetch (file:///src/main.jsx:35:17)
    at handler (ext:deno_http/00_serve.ts:652:26)
    at ext:deno_http/00_serve.ts:369:24
    at ext:deno_http/00_serve.ts:569:29
    at eventLoopTick (ext:core/01_core.js:175:7) {
  name: "HttpError",
  status: 404
}

**Reply by johnd0e** [reply]
The error is raised for unsupported urls.
Should be no problem when fetching OpenAI-style endpoints.
